hal/halx86/smp/mps/mps.c

```diff
@@ -0,0 +1,41 @@
+
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Source File MPS specific functions
+ * PROGRAMMERS: Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <hal.h>
+#include <smp.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+PROCESSOR_IDENTITY HalpStaticProcessorIdentity[MAX_CPUS] = {{0}};
+PPROCESSOR_IDENTITY HalpProcessorIdentity = NULL;
+UINT32 PhysicalProcessorCount = 0;
+
+/* FUNCTIONS ******************************************************************/
+
+VOID
+HalpParseApicTables(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
+{
+    UNIMPLEMENTED;
+}
+
+VOID
+HalpPrintApicTables()
+{ 
+    DPRINT1("HAL has detected a physical processor count of: %d\n", PhysicalProcessorCount);
+
+    for (UINT32 i = 0; i < PhysicalProcessorCount; i++)
```

Comment:
please move the declaration of i to the top of the function.

Reply:
To note again: This is Joachim's personal preference, and is not supported by the entire team.